staging/vhost-device-gpu/src/protocol.rs

```diff
@@ -0,0 +1,1261 @@
+        assert_eq!(
+            format!("{:?}", set_scanout_blob),
+            expected_debug_output_scanout_blob
+        );
+        assert_eq!(format!("{:?}", resource_flush), expected_debug_output_flush);
+        assert_eq!(
+            format!("{:?}", transfer_to_host_2d),
+            expected_debug_output_transfer_to_host_2d
+        );
+        assert_eq!(
+            format!("{:?}", resource_detach_backing),
+            expected_debug_output_detach_backing
+        );
+        assert_eq!(
+            format!("{:?}", get_capset_info),
+            expected_debug_output_get_capset_info
+        );
+        assert_eq!(
+            format!("{:?}", get_capset),
+            expected_debug_output_get_capset
+        );
+        assert_eq!(
+            format!("{:?}", ctx_create),
+            expected_debug_output_ctx_create
+        );
+        assert_eq!(
+            format!("{:?}", ctx_destroy),
+            expected_debug_output_ctx_destroy
+        );
+        assert_eq!(
+            format!("{:?}", ctx_attach_resource),
+            expected_debug_output_ctx_attach_resource
+        );
+        assert_eq!(
+            format!("{:?}", ctx_detach_resource),
+            expected_debug_output_ctx_detach_resource
+        );
+        assert_eq!(
+            format!("{:?}", resource_create_3d),
+            expected_debug_output_resource_create_3d
+        );
+        assert_eq!(
+            format!("{:?}", transfer_to_host_3d),
+            expected_debug_output_transfer_to_host_3d
+        );
+        assert_eq!(
+            format!("{:?}", transfer_from_host_3d),
+            expected_debug_output_transfer_from_host_3d
+        );
+        assert_eq!(
+            format!("{:?}", cmd_submit_3d),
+            expected_debug_output_cmd_submit_3d
+        );
+        assert_eq!(
+            format!("{:?}", resource_create_blob),
+            expected_debug_output_create_blob
+        );
+        assert_eq!(
+            format!("{:?}", resource_map_blob),
+            expected_debug_output_map_blob
+        );
+        assert_eq!(
+            format!("{:?}", resource_unmap_blob),
+            expected_debug_output_unmap_blob
+        );
+        assert_eq!(
+            format!("{:?}", update_cursor),
+            expected_debug_output_update_cursor
+        );
+        assert_eq!(
+            format!("{:?}", move_cursor),
+            expected_debug_output_move_cursor
+        );
+        assert_eq!(
+            format!("{:?}", resource_assign_uuid),
+            expected_debug_output_assign_uuid
+        );
+
+        let bytes = b"test_debug\0";
+        let original = virtio_gpu_ctx_create {
+            debug_name: {
+                let mut debug_name = [0; 64];
+                debug_name[..bytes.len()].copy_from_slice(bytes);
+                debug_name
+            },
+            context_init: 0,
+            nlen: bytes.len() as u32,
```

Comment:
Btw fyi this includes the `\0` NUL character. C would expect a `strlen(_)` result. **Doesn't make difference** here if it's bytes.len() or bytes.len() minus one at all but it's good to have in mind generally that when doing C FFI with NULL terminators to adjust the lengths.

Reply:
I added this as a comment, not a review, I don't know why it says "resolve this" :B Feel free to click it if you read this.